config/role.yaml

```diff
@@ -22,43 +109,86 @@ rules:
   - namespaces
   verbs:
   - get
+  - update
+  - watch
+- apiGroups:
+  - ''
+  resources:
+  - events
+  verbs:
+  - create
+  - update
+  - patch
+- apiGroups:
+  - ''
+  resources:
+  - configmaps
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - watch
 - apiGroups:
   - apps
   resources:
   - deployments
-  - daemonsets
-  - replicasets
-  - statefulsets
   verbs:
-  - '*'
+  - create
+  - delete
+  - get
+  - list
+  - watch
 - apiGroups:
-  - monitoring.coreos.com
+  - apiregistration.k8s.io
   resources:
-  - servicemonitors
+  - apiservices
   verbs:
+  - create
+  - delete
   - get
+  - list
+- apiGroups:
+  - autoscaling
+  resources:
+  - horizontalpodautoscalers
+  verbs:
   - create
+  - delete
+  - get
+  - list
+# Permissions required specifically for this Operator.
 - apiGroups:
-  - apps
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - '*'
+- apiGroups:
+  - ''
+  resources:
+  - nodes
   resourceNames:
-  - knative-serving-operator
+  - minikube
+  verbs:
+  - get
+# Old resources that need cleaning up that are not in the knative-serving
```

Comment:
When can these be deleted?

Reply:
Whenever the Operator code to clean up the resources can be, it's not clear to me what our strategy is in that regard.

I imagine it will become clear once we have clear upgrade/downgrade and versioning story between the operator what it installs :)

Reply:
See manually curated comment chain.